test/commands/code-analyzer/rules.test.ts

```diff
@@ -101,13 +101,17 @@ describe('`code-analyzer rules` tests', () => {
 			await RulesCommand.run(['--view', inputValue]);
 			expect(spy).toHaveBeenCalled();
 			expect(receivedActionInput).toHaveProperty('view', inputValue);
+			// TODO: Janky workaround. Can we replace with actual mockup of constructor or similar?
+			expect(spy.mock.contexts[0].dependencies.viewer.constructor.name).toEqual('RuleTableViewer');
```

Comment:
I'm highly aware of how janky this pattern is. In any reasonable world, I'd spy on the constructor for `RulesAction` and make sure that the right type of Viewer is passed in then. But I spent multiple hours trying and failing to make that happen, and this worked. So for now, this is what I'm going with.

Reply:
See my other comment about createing an overridable createAction method on your RulesCommand that your test can override with the stubbed out dependencies.